### ISSUE TITLE
EIS 2023 calibration 

### ISSUE THREAD

**Issue by andyto1234** [USNavalResearchLaboratory/eispac#91]
Sorry this might be very dumb on my side.

I briefly remembers the eispac code uses Warren et al. 2014 to calibrate the files? Is there a place this is specified? I think adding the ability/tutorial to use different calibrations (e.g. the latest Del Zanna et al. 2023) could be extremely useful.

**Comment by MJWeberg** [comment]
Hello Andy,
This is very good question! EISPAC actually currently only has the pre-flight calibration available, unfortunately. However, the one of the goals is to eventually have the level-1 HDF5 files serve as a long-term archive of EIS data and associated metadata. As such, we are planning to update the files in the next few months to include all of the the calibrations currently available. The 2023 calibration is still going through peer-review, so we are waiting a little bit to make sure we do not need to revise or update the methodology further before we reprocess all the header files. Sorry it has taken so long!

**Comment by andyto1234** [comment]
Hi Micah!
Thank you for the clear reply! :D  Let me know if I can be of any help!

**Comment by andyto1234** [comment]
Thought I will keep the thread updated with this.

 I translated the IDL calibrations (Warren et al. 2014 and Del Zanna et al. 2023 in review) into Python (pretty much a 1 to 1 translation). You can find the tutorial in the following link.

https://github.com/andyto1234/EISPAC-Tutorial___Calibrations/

**Comment by souvikita** [comment]
This is very helpful, Andy! Thanks!

**Comment by wtbarnes** [comment]
@MJWeberg Have the files been updated to include the new calibration or is still necessary to apply this correction after the fact (e.g. as Andy has done)?

**Comment by MJWeberg** [comment]
Thanks for following up on this! Updating all of the HDF5 files to include all of the available calibrations is near the top of our TO-DO list (it has been a looong process). Applying the calibration after fitting is fine, since the calibration generally does not change on scales small enough to affect the fitting results. Andy's very helpful functions are direct ports of the SSW routines we wrote/updated as part of the recent calibration and I have not noticed any notable differences myself (I should probably just run some formal validation tests and work with Andy to include these functions directly in EISPAC).

**Comment by hpwarren** [comment]
Sorry for taking so long to add my 2 cents . . . Another nuance: Peter did
a very interesting of a full CCD spectrum taken during a flare. There is
generally good agreement with the curves in the paper. The notable
exception is the region around 265 Ã. The flare spectrum doesn't seem to be
consistent with the ripple indicated by the ratio analysis.

Harry

**Comment by wtbarnes** [comment]
> (I should probably just run some formal validation tests and work with Andy to include these functions directly in EISPAC).

Yes I think that would be great! Even after the most recent calibration is applied to the L1 files, it would be great to have the ability to apply different calibrations to the resulting maps.
